R/bambu-extendAnnotations-utilityExtend.R

```diff
@@ -548,74 +547,16 @@ assignGeneIDexonMatch <- function(rowDataCombined, exonRangesCombined,
     geneIdByExon[isNaGeneIdBefore]
   isNaGeneIdAfter <- is.na(rowDataCombined$GENEID)
   if (any(isNaGeneIdAfter)) {
-    newGeneIds <- 
-      assignNewGeneIds(exonRangesCombined[isNaGeneIdAfter],
-                       prefix = prefix, minoverlap = 5, ignore.strand = FALSE)
-    rowDataCombined$GENEID[as.integer(newGeneIds$readClassId)] <- 
-      newGeneIds$geneId
+    newGeneIds <- assignGeneIdsNoReference(exonRangesCombined[isNaGeneIdAfter])
+    rowDataCombined$GENEID[isNaGeneIdAfter] <- 
+      newGeneIds 
   }
   end.ptm <- proc.time()
   if (verbose) message("assigned read classes to annotated and
         new gene IDs in ", round((end.ptm - start.ptm)[3] / 60, 1), " mins.")
   return(rowDataCombined)
 }
 
-
-#' Assign New Gene with Gene Ids
-#' @param exByTx exByTx
-#' @param prefix prefix, defaults to empty
-#' @param minoverlap defaults to 5
-#' @param ignore.strand defaults to FALSE
-#' @importFrom dplyr as_tibble %>% filter select ungroup inner_join mutate
-#'     distinct if_else arrange tibble
-#' @noRd
-assignNewGeneIds <- function(exByTx, prefix = "", minoverlap = 5,
```

Comment:
I think this might be the only function that uses "prefix"? we can remove prefix from the argument list then?

Reply:
bfec4850170b708dcf28bbf18743ec8b99da970c
At first i was going to remove prefixes but I realized there is a potential use case where a user might rerun bambu with bambu annotations which would mean the gene and tx ids would be duplicated and cause issues in the gtf file. By providing a unique prefix when running bambu it stops that duplication from occuring.
Therefore I returned the prefix to assignGeneIdsByReference(). This should have no functional change for regular use.